src/components/menus/directory-tree-contextual-menu-controller.js

```diff
@@ -0,0 +1,312 @@
+        onClose(e, nextSelectedDirectoryId);
+        setOpenDialog(DialogsId.NONE);
+        setLastError('');
+        setHideMenu(false);
+    };
+
+    /* Handle Dialogs actions */
+    function insertNewDirectory(directoryName, isPrivate) {
+        insertDirectory(
+            directoryName,
+            directory?.elementUuid,
+            isPrivate,
+            userId
+        ).then((newDir) => {
+            handleCloseDialog(null, newDir.elementUuid);
+        });
+    }
+
+    function insertNewRootDirectory(directoryName, isPrivate) {
+        insertRootDirectory(directoryName, isPrivate, userId).then((newDir) => {
+            handleCloseDialog(null, newDir.elementUuid);
+        });
+    }
+
+    function deleteSelectedDirectory() {
+        deleteElement(directory?.elementUuid).then((r) => {
+            if (r.ok) {
+                handleCloseDialog(null, directory.parentUuid);
+            }
+            if (r.status === 403) {
+                setLastError(
+                    intl.formatMessage({ id: 'deleteDirectoryError' })
+                );
+            }
+        });
+    }
+
+    function changeSelectedDirectoryAccessRights(isPrivate) {
+        updateAccessRights(directory?.elementUuid, isPrivate).then((r) => {
+            if (r.status === 403) {
+                setLastError(
+                    intl.formatMessage({
+                        id: 'modifyDirectoryAccessRightsError',
+                    })
+                );
+            }
+            if (r.ok) {
+                handleCloseDialog(null, null);
+            }
+        });
+    }
+
+    function renameSelectedDirectory(newName) {
+        renameElement(directory?.elementUuid, newName).then((r) => {
+            if (r.status === 403) {
+                setLastError(
+                    intl.formatMessage({
+                        id: 'renameDirectoryError',
+                    })
+                );
+            }
+            if (r.ok) {
+                handleCloseDialog(null, null);
+            }
+        });
+    }
+
+    // utils
+
+    // Allowance
+    const showMenuFromEmptyZone = useCallback(() => {
+        return !directory;
+    }, [directory]);
+
+    const isAllowed = useCallback(() => {
+        return directory && directory.owner === userId;
+    }, [directory, userId]);
+
+    const renderMenu = () => {
+        // build menuItems here
+        let menuItems = [];
+
+        if (!showMenuFromEmptyZone()) {
+            menuItems.push({
+                messageDescriptorId: 'createNewStudy',
+                callback: () => {
+                    setHideMenu(true);
+                    handleOpenDialog(DialogsId.ADD_NEW_STUDY);
+                },
+                icon: <AddIcon fontSize="small" />,
+            });
+
+            menuItems.push({
+                messageDescriptorId: 'createNewContingencyList',
+                callback: () => {
+                    setHideMenu(true);
```

Comment:
can setHideMenu be put in handleOpenDialog ?

Reply:
IT must and it does now ! :)